### ISSUE TITLE
Cppcheck shows all messages as errors

### ISSUE THREAD

**Issue by janek202** [jbenden/vscode-c-cpp-flylint#110]
Hello,

I have a problem with the Cppcheck linter.
All messages, at all severity levels, are shown as errors.

For example:

```c++
struct foo {
    virtual void baz() {}
};

struct bar : public foo {
    void baz() {}
};
```
Shows "style" as an error:
```json
{
	"resource": "(...)/cpp_scratch/lint.cpp",
	"owner": "_generated_diagnostic_collection_name_#1",
	"code": "missingOverride",
	"severity": 8,
	"message": "The function 'baz' overrides a function in a base class but is not marked with a 'override' specifier.",
	"source": "CppCheck",
	"startLineNumber": 6,
	"startColumn": 5,
	"endLineNumber": 6,
	"endColumn": 18
}
```

I'm using the latest VS Code (Version: 1.53.2) and Cppcheck 2.3, on a Mac. 
But I have the same problem with Linux hosts accessed over the SSH Remote.
I don't have any custom settings for this extension, other than disabling Flawfinder, Flexelint and Lizard.

Is there a problem in the latest version of the extension and/or VS Code?
Do I have to configure something?

Regards,
Jan

**Comment by jbenden** [comment]
Hi, thanks for the bug report!

There is some sort of a problem in mapping cppcheck's severity codes. Additionally, it is somehow the fault of TypeScript in casting values. Odd.

I changed the code around to be explicit in conversions, which (locally) fixes the problems. The fix will appear in version 1.8.1.

Thanks,
-Joe

**Comment by janek202** [comment]
Hi,

Thank you for a quick fix!
Version 1.8.1 fixes the issue for me, on all my hosts.

Regards,
Jan
